server/swap/swap_test.go

```diff
@@ -823,11 +816,13 @@ func (rig *testRig) sendSwap(user *tUser, oid order.OrderID, recipient string) (
 	} else {
 		rig.abcNode.setContract(swap.coin, false)
 	}
+	rig.auth.mtx.Lock()
 	rig.auth.swapID = swap.req.ID
+	rig.auth.mtx.Unlock()
```

Comment:
Also seems like there is a race with `TAuthManager.swapID` field here (and this ^ should fix it):
```
WARNING: DATA RACE
Write at 0x00c0003c1cd0 by goroutine 97:
  decred.org/dcrdex/server/swap.(*testRig).sendSwap()
      /Users/iurii/dcrdex/server/swap/swap_test.go:829 +0x2f9
  decred.org/dcrdex/server/swap.(*testRig).sendSwap_maker()
      /Users/iurii/dcrdex/server/swap/swap_test.go:765 +0x127
  decred.org/dcrdex/server/swap.TestRetrySwapInit()
      /Users/iurii/dcrdex/server/swap/swap_test.go:2045 +0x2de
  testing.tRunner()
      /Users/iurii/go/src/testing/testing.go:1259 +0x22f
  testing.(*T).Run·dwrap·21()
      /Users/iurii/go/src/testing/testing.go:1306 +0x47

Previous read at 0x00c0003c1cd0 by goroutine 42:
  decred.org/dcrdex/server/swap.(*TAuthManager).Send()
      /Users/iurii/dcrdex/server/swap/swap_test.go:159 +0x3ac
  decred.org/dcrdex/server/swap.(*Swapper).respondSuccess()
      /Users/iurii/dcrdex/server/swap/swap.go:1258 +0x1c3
  decred.org/dcrdex/server/swap.(*Swapper).processInit()
      /Users/iurii/dcrdex/server/swap/swap.go:1631 +0x2345
  decred.org/dcrdex/server/swap.(*Swapper).handleInit.func1()
      /Users/iurii/dcrdex/server/swap/swap.go:1956 +0x6b
  decred.org/dcrdex/dex/wait.(*TaperingTickerQueue).Run.func1()
      /Users/iurii/dcrdex/dex/wait/queue.go:197 +0xfa
  decred.org/dcrdex/dex/wait.(*TaperingTickerQueue).Run·dwrap·5()
      /Users/iurii/dcrdex/dex/wait/queue.go:248 +0x47
```

Reply:
Maybe another test already had the same issue, but `TAuthManager.swapID` being modified while there is as swap presently being handled with `handleInit` doesn't make sense.  (If a subsequent `sendSwap` modified it while a previous `handleInit` was already running, then the `Send` for the response on the first one would reference the wrong `swapID`! Thus the test rig won't work right unless the swapID happens to be the same like it is in your test).

In other words, this `swapID` field is just a single value because the test rig isn't designed to work properly with multiple concurrent `handleInit` calls, and all of the prior tests respected that.

I think you're only seeing this race now because of what `TestRetrySwapInit` is trying to test.
OK to locking in `sendSwap`, but I don't think other tests were hitting this race, and any that do including this new one are potentially in error.

Reply:
Revised this a bit, I think I was observing this race while `TestRetrySwapInit` wasn't "complete", now it can locking on this mutex can be safely removed I think (done).